File: 3056-determine-if-a-cell-is-reachable-at-a-given-time/determine-if-a-cell-is-reachable-at-a-given-time.py
class Solution(object):
    def isReachableAtTime(self, sx, sy, fx, fy, t):
        dist_x = abs(fx - sx)
        dist_y = abs(fy - sy)
        minimum_dist = min(dist_x, dist_y)
        rem_x = dist_x - minimum_dist
        rem_y = dist_y - minimum_dist
        if max(rem_x, rem_y) + minimum_dist == 0:
            if t != 1:
                return True
            else:
                return False
        return max(rem_x, rem_y) + minimum_dist <= t

        

File: 3056-determine-if-a-cell-is-reachable-at-a-given-time/determine-if-a-cell-is-reachable-at-a-given-time.java
class Solution {
    public boolean isReachableAtTime(int sx, int sy, int fx, int fy, int t) {
        int dist_x = Math.abs(fx - sx), dist_y = Math.abs(fy - sy);
        int minimum_dist = Math.min(dist_x, dist_y);
        int rem_x = dist_x - minimum_dist, rem_y = dist_y - minimum_dist;
        if(Math.max(rem_x, rem_y) + minimum_dist == 0){
            if(t != 1){
                return true;
            }else{
                return false;
            }
        }
        return Math.max(rem_x, rem_y) + minimum_dist <= t;
    }
}
